### DISCUSSION TITLE
Version dependencies between torch-lightning and torch.

### DISCUSSION THREAD

**Discussion by HubertSing** [Lightning-AI/pytorch-lightning#14743] [General]
Hello, I would like to ask a question about the version dependencies between torch-lightning and torch.
Because my GPU can only support cuda 10.1, so my pytorch version can only be up to 1.8.
In order to maintain the project in the long term, I would  likely to be able to use torch-lightning in a long-term and convenient way.
However, the one-click installation command on the official website will directly overwrite my torch version to 1.12, and the version of cudatoolkit will also become higher, which makes my environment dependencies wrong and I had to reconfigure the environment.
After searching in the issues section of github, I found that I should use the pip install pytorch-lightning==1.6.* command.
I'm curious as to where to get the full compatibility between previous versions of pytorch-lightning and torch.
Any help would be greatly appreciated, thanks!

**Comment by justusschock** [comment]
Hey, 

we try to always support the last 3 major releases of torch (currently being 1.12, 1.11 and 1.10). Everything beyond that might work in most cases but isn't tested and we cannot guarantee stability of support of all features.

This decision was done out of two reasons:
- Our CI would need to have a lot more jobs so testing might become very hard
- We basically would have a hard time moving forward if we wouldn't narrow down the versions. It already is hard to implement features only for specific versions with the existing supported bandwidth and to properly take care of deprecations but since PyTorch does move increasingly fast, we would basically face blockers way to often to make this feasible. 

That being said, If you really need this, try to install it like that:

`pip install pytorch-lightning pytorch==1.8` (it shouldn't update the requirement then and use the manually specified torch version)

but as I said, we cannot guarantee anything that way, so proceed at your own risk .

**Reply by HubertSing** [reply]
That's enough for me!
Thanks for your timely answer!

**Reply by akihironitta** [reply]
FYI, it's recently been documented at https://pytorch-lightning.readthedocs.io/en/latest/versioning.html#pytorch-support.
